src/obu.rs

```diff
@@ -1930,63 +1928,43 @@ unsafe fn parse_obus(
         if c.frame_hdr.is_null() {
             return Err(EINVAL);
         }
-        if c.n_tile_data_alloc < c.n_tile_data + 1 {
-            if c.n_tile_data + 1
-                > i32::MAX / ::core::mem::size_of::<Rav1dTileGroup>() as c_ulong as c_int
-            {
-                return Err(EINVAL);
-            }
-            let tile = realloc(
-                c.tile as *mut c_void,
-                ((c.n_tile_data + 1) as usize) * ::core::mem::size_of::<Rav1dTileGroup>(),
-            ) as *mut Rav1dTileGroup;
-            if tile.is_null() {
-                return Err(EINVAL);
-            }
-            c.tile = tile;
-            memset(
-                c.tile.offset(c.n_tile_data as isize) as *mut c_void,
-                0,
-                ::core::mem::size_of::<Rav1dTileGroup>(),
-            );
-            c.n_tile_data_alloc = c.n_tile_data + 1;
-        }
-        let tile_hdr = parse_tile_hdr(&(*c.frame_hdr).tiling, gb);
-        (*(c.tile).offset(c.n_tile_data as isize)).hdr = tile_hdr;
+
+        let hdr = parse_tile_hdr(&(*c.frame_hdr).tiling, gb);
         // Align to the next byte boundary and check for overrun.
         rav1d_bytealign_get_bits(gb);
         if check_for_overrun(c, gb, init_bit_pos, len) != 0 {
             return Err(EINVAL);
         }
+
+        if let Err(_) = c.tiles.try_reserve_exact(1) {
+            return Err(EINVAL);
+        }
+        c.tiles.push(Rav1dTileGroup {
+            data: Default::default(),
+            hdr,
+        });
+        // TODO(kkysen) An idiom for push and getting a &mut would be nice and more efficient.
```

Comment:
Why not move the push to after the initialization? I don't think we need a new api for this.

Reply:
I thought I did this for a reason; I think I thought `rav1d_data_ref` had some internal pointer or something, but I think I was wrong, so I should be able to do it this way.  It's definitely a lot cleaner.

Reply:
Fixed in 349b43c.